cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/MasterEnvironmentMain.java

```diff
@@ -102,15 +104,23 @@ public static void doMain(String[] args) throws Exception {
       }
 
       CConfiguration cConf = CConfiguration.create();
+      File cConfFile = new File(CONFIGMAP_MOUNTPATH, "cConf.xml");
+      if (cConfFile.exists()) {
+        cConf.addResource(cConfFile.toURI().toURL());
+      }
+
       SConfiguration sConf = SConfiguration.create();
       if (options.getExtraConfPath() != null) {
-        cConf.addResource(new File(options.getExtraConfPath(), "cdap-site.xml").toURI().toURL());
```

Comment:
does this mean this extra conf path is never used (I think we just mount it to /etc/cdap/conf in every container)? If so we should just remove it everywhere

Reply:
Seems to be unused except in MasterServiceMainTestBase - https://github.com/cdapio/cdap/blob/develop/cdap-master/src/test/java/io/cdap/cdap/master/environment/k8s/MasterServiceMainTestBase.java#L244.
I don't think this is actually used though. Is it safe to remove?

Reply:
I see. We don't want to get rid of that test, and we don't want to write to hardcoded paths in the unit test.

Let's change that test to just write the files as cConf.xml/sConf.xml instead of cdap-site.xml and cdap-security.xml. Instead of hardcoding the config path to /config here, the KubeTwillPreparer should set the conf path option to where it will be mounting the files.

Reply:
Actually I see that AbstractServiceMain specifies filenames cdap-site.xml, cdap-security.xml, so I left the test as is - https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/master/environment/k8s/AbstractServiceMain.java#L149.

I did fix KubeTwillPreparer to set the conf path option as you suggested, and MasterEnvironmentMain now copies config files from the extraConfPath (doesn't hardcode /config).